src/browserExt/mimeTypeHandler.js

```diff
@@ -0,0 +1,181 @@
+	},
+
+	observe: function(details) {
+		if (!details.responseHeadersObject['content-type']) return;
+		if (Zotero.MIMETypeHandler.ignoreURL.has(details.url)) {
+			// Ignore url when user cancels 
+			Zotero.MIMETypeHandler.ignoreURL.delete(details.url);
+			return;
+		}
+		
+		let URI = url.parse(details.url);
+		let contentType = details.responseHeadersObject['content-type'].split(';')[0];
+		if (Zotero.MIMETypeHandler.cslMIMETypes.has(contentType)) {
+			Zotero.MIMETypeHandler.confirm(details, `Would you like Zotero to install a style from '${details.url}'?`)
+				.then(function(response) {
+					if (response) {
+						Zotero.MIMETypeHandler.importFile(details, 'csl');
+					}
+				}
+			);
+			// Don't continue until we get confirmation
+			return {redirectUrl: 'javascript:'};
+		} else if (Zotero.Prefs.get('interceptAndImport') && 
+			Zotero.MIMETypeHandler.importMIMETypes.has(contentType)) {
+			
+			let hosts = Zotero.Prefs.get('allowedInterceptHosts');
+			let isEnabledHost = hosts.indexOf(URI.host) != -1;
+			if (isEnabledHost) {
+				Zotero.MIMETypeHandler.importFile(details, 'import');
+			} else {
+				Zotero.MIMETypeHandler.confirm(details, `Would you like to use Zotero to import items from ${URI.host}?\n\n` +
+					'You can manage automatic file import in Zotero Connector preferences.').then(function(response) {
+					if (response) {
+						Zotero.MIMETypeHandler.importFile(details, 'import');
+						if (!isEnabledHost) {
+							hosts.push(URI.host);
+						}
+						Zotero.Prefs.set('allowedInterceptHosts', hosts);
+					}
+				});
+			}
+			return {redirectUrl: 'javascript:'};
+		}
+	},
+
+	/**
+	 * Handle confirmation prompt by sending a message to injected script and
+	 * redirect to the URL if they click cancel
+	 * @param details {Object} as provided by Zotero.WebRequestsInterceptor
+	 * @param message {String} confirmation message to display
+	 * @returns {Promise{Boolean}} whether user clicked OK or Cancel
+	 */
+	confirm: function(details, message) {
+		let deferred = Zotero.Promise.defer();
+		chrome.tabs.get(details.tabId, function(tab) {
+			Zotero.Messaging.sendMessage('confirm', message, tab).then(function(response) {
+				if (!response) {
+					Zotero.MIMETypeHandler.ignoreURL.add(details.url);
+					// Ignore the next request to this url and redirect
+					chrome.tabs.update(tab.id, {url: details.url});
+				}
+				deferred.resolve(response);
+			});	
+		});	
+		
+		return deferred.promise;
+	},
+	
+	/**
+	 * Send an XHR request to retrieve and import the file into Standalone
+	 */
+	importFile: function(details, type) {
+		chrome.tabs.get(details.tabId, function(tab) {
+			// Make sure scripts injected so we can display the progress window
+			Zotero.Connector_Browser.injectScripts(tab.id).then(function() {
+				Zotero.Messaging.sendMessage('progressWindow.show', type == 'csl' ? 'Installing a style' : null, tab);
+			
+				var xhr = new XMLHttpRequest();
+				xhr.open('GET', details.url);
```

Comment:
Oh, yeah, that wouldn't be good. Can we determine from here whether the original request was a POST, and not show the dialog if so? There's no reason to think a GET would work for a POST, and if we're retriggering the request on Cancel too (I'm not sure if we are) then this would mean breaking download of such files.